prow/cmd/crier/main.go

```diff
@@ -0,0 +1,132 @@
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/time/rate"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+
+	prowjobclientset "k8s.io/test-infra/prow/client/clientset/versioned"
+	prowjobinformer "k8s.io/test-infra/prow/client/informers/externalversions"
+
+	"k8s.io/test-infra/prow/crier"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+const (
+	resync = 0 * time.Second
+)
+
+// TODO(krzyzacy): copy & paste, refactor this
+// loadClusterConfig loads connection configuration
+// for the cluster we're deploying to. We prefer to
+// use in-cluster configuration if possible, but will
+// fall back to using default rules otherwise.
+func loadClusterConfig() (*rest.Config, error) {
+	clusterConfig, err := rest.InClusterConfig()
+	if err == nil {
+		return clusterConfig, nil
+	}
+
+	credentials, err := clientcmd.NewDefaultClientConfigLoadingRules().Load()
+	if err != nil {
+		return nil, fmt.Errorf("could not load credentials from config: %v", err)
+	}
+
+	clusterConfig, err = clientcmd.NewDefaultClientConfig(*credentials, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load client configuration: %v", err)
+	}
+	return clusterConfig, nil
+}
+
+// TODO(krzyzacy): move this to kube???
+// getKubernetesClient retrieves the Kubernetes cluster
+// client from within the cluster
+func getKubernetesClient() (kubernetes.Interface, prowjobclientset.Interface) {
+	config, err := loadClusterConfig()
+	if err != nil {
+		logrus.Fatalf("failed to load cluster config: %v", err)
+	}
+
+	// generate the client based off of the config
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logrus.Fatalf("getClusterConfig: %v", err)
+	}
+
+	prowjobClient, err := prowjobclientset.NewForConfig(config)
+	if err != nil {
+		logrus.Fatalf("getClusterConfig: %v", err)
+	}
+
+	logrus.Info("Successfully constructed k8s client")
+	return client, prowjobClient
+}
+
+func main() {
+	logrus.SetFormatter(
+		logrusutil.NewDefaultFieldsFormatter(nil, logrus.Fields{"component": "crier"}),
+	)
+
+	// get the Kubernetes client for connectivity
+	client, prowjobClient := getKubernetesClient()
+
+	prowjobInformerFactory := prowjobinformer.NewSharedInformerFactory(prowjobClient, resync)
+
+	// create a new queue so that when the informer gets a resource that is either
+	// a result of listing or watching, we can add an idenfitying key to the queue
+	// so that it can be handled in the handler
+	queue := workqueue.NewRateLimitingQueue(
+		workqueue.NewMaxOfRateLimiter(
+			workqueue.NewItemExponentialFailureRateLimiter(1*time.Second, 60*time.Second),
+			// 10 qps, 100 bucket size.  This is only for retry speed and its only the overall factor (not per item)
+			&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(10), 100)},
+		))
+
+	// construct the Controller object which has all of the necessary components to
```

Comment:
Not sure we really need these comments? Are they from the sample code?

Reply:
cleaned up the comments from sample code, also cleaned up logs